tests/basic_functionality_test.py

```diff
@@ -33,3 +43,79 @@ def test_has_link_layer() -> None:
             break
 
     assert has_any_link, "Test failure; no AF_PACKET address of any kind found"
+
+
+@pytest.mark.skipif(
+    platform.system() != "Windows", reason="Windows only"
+)  # type: ignore[misc]
+def test_interface_display_formats_windows() -> None:
+    """
+    Check that the InterfaceDisplay argument can be used to select between a UUID
+    and a human readable name
+    """
+
+    uuid_regex = r"{[-A-F0-9]+}"
+
+    # The machine readable interface should look like a UUID string
+    machine_readable_iface0 = netifaces.interfaces(
+        netifaces.InterfaceDisplay.MachineReadable
+    )[0]
+    print(f"Machine readable name of interface 0 is: {machine_readable_iface0}")
+    assert re.fullmatch(uuid_regex, machine_readable_iface0) is not None
+
+    # The human readable interface should NOT look like a UUID
+    human_readable_iface0 = netifaces.interfaces(
+        netifaces.InterfaceDisplay.HumanReadable
+    )[0]
+    print(f"Human readable name of interface 0 is: {human_readable_iface0}")
+    assert re.fullmatch(uuid_regex, human_readable_iface0) is None
+
+
+def test_loopback_addr_is_returned() -> None:
+    """
+    Test that the loopback address is returned in the lists of addresses
+    (regression test for a bug)
+    """
+
+    loopback_ipv4_found = False
+    loopback_ipv6_found = False
+
+    for interface in netifaces.interfaces():
+        address_table = netifaces.ifaddresses(interface)
+
+        if netifaces.AF_INET in address_table:
+            for ipv4_settings in address_table[netifaces.InterfaceType.AF_INET]:
+                print(
+                    f"Loopback test: Considering iface {interface} IPv4 address "
+                    f"{ipv4_settings['addr']}"
+                )
+                if ipv4_settings["addr"] == "127.0.0.1":
+                    print("Loopback IPv4 found!")
+                    loopback_ipv4_found = True
+
+        if netifaces.AF_INET6 in address_table:
+            for ipv6_settings in address_table[netifaces.InterfaceType.AF_INET6]:
```

Comment:
There's a known issue on windows where the AF_INET values are not correct on windows (they are platform dependent, so I am thinking of using magic values for all of them, since it's somewhat pointless to use the platform's value)

Reply:
(FYI)

Reply:
Actually I fixed the issue with AF_INET6 on Windows in this PR